pkg/apis/autoscaling/register.go

```diff
@@ -17,15 +17,20 @@ limitations under the License.
 package autoscaling
 
 const (
-	GroupName = "autoscaling.internal.knative.dev"
+	InternalGroupName = "autoscaling.internal.knative.dev"
+
+	GroupName = "autoscaling.knative.dev"
 
 	// ClassAnnotationKey is the annotation for the explicit class of autoscaler
 	// that a particular resource has opted into. For example,
 	//   autoscaling.knative.dev/class: foo
 	// This uses a different domain because unlike the resource, it is user-facing.
-	ClassAnnotationKey = "autoscaling.knative.dev/class"
+	ClassAnnotationKey = GroupName + "/class"
+
+	MinScaleAnnotationKey = GroupName + "/minReplicaCount"
+	MaxScaleAnnotationKey = GroupName + "/maxReplicaCount"
```

Comment:
I feel like we should define these in `pkg/autoscaler`, not alongside our types (continuation of the comment on `spec.md`)

Reply:
👍  to @mattmoor.

IMHO honoring `ResourceQuota` specs for the given namespace of the `ReplicaSet` wrt `maxReplicaCount` might be very wise a design.

RFC @mattmoor  @imikushin 🙏

Reply:
@mattmoor The issue this PR is addressing (#1656) suggests the placement and rationale for these annotations:

> I would like to add support for two KPA annotations ([here](https://github.com/knative/serving/blob/928d580756855f24d400104ad4d4cc628a8a841f/pkg/apis/autoscaling/register.go#L26)) to constrain the min and max number of pods.
> ```
> autoscaling.knative.dev/minReplicaCount
> autoscaling.knative.dev/maxReplicaCount
> ```
> The annotations would be in the user-facing domain (not autoscaling.internal.knative.dev) because they are intended to be set by the Operator to place bounds on resource consumption.
> 
> They would also apply to other autoscaling implementations specified by autoscaling.knative.dev/class (if the implementation supports min/max constraints.)
> 

@nrshrivatsan ResourceQuotas address a similar but different concern IMO and they apply to the whole namespace. Scale bounds only apply to specific revisions and enable both upper and _lower_ bounds (kinda opposite to quotas' purpose).